lib/ansible/modules/cloud/azure/azure_rm_trafficmanager.py

```diff
@@ -0,0 +1,580 @@
+description:
+    - Create, update and delete a Traffic Manager profile.
+
+options:
+    resource_group:
+        description:
+            - Name of a resource group where the Traffic Manager profile exists or will be created.
+        required: true
+    name:
+        description:
+            - Name of the Traffic Manager profile.
+        required: true
+    state:
+        description:
+            - Assert the state of the Traffic Manager profile. Use C(present) to create or update a Traffic Manager profile and C(absent) to delete it.
+        default: present
+        choices:
+            - absent
+            - present
+    location:
+        description:
+            - Valid azure location. Defaults to 'global'.
+    profile_status:
+        description:
+            - The status of the Traffic Manager profile.
+        default: Enabled
+        choices:
+            - Enabled
+            - Disabled
+    traffic_routing_method:
+        description:
+            - The traffic routing method of the Traffic Manager profile.
+        default: Performance
+        choices:
+            - Performance
+            - Priority
+            - Weighted
+            - Geographic
+    dns_config:
+        description:
+            - The DNS settings of the Traffic Manager profile.
+        suboptions:
+            relative_name:
+                description:
+                    - The relative DNS name provided by this Traffic Manager profile.
+                    - If no provided, name of the Traffic Manager will be used
+            ttl:
+                description:
+                    - The DNS Time-To-Live (TTL), in seconds.
+                default: 60
+    monitor_config:
+        description:
+            - The endpoint monitoring settings of the Traffic Manager profile.
+        suboptions:
+            protocol:
+                description:
+                    - The protocol (HTTP, HTTPS or TCP) used to probe for endpoint health.
+                choices:
+                    - HTTP
+                    - HTTPS
+                    - TCP
+            port:
+                description:
+                    - The TCP port used to probe for endpoint health.
+            path:
+                description:
+                    - The path relative to the endpoint domain name used to probe for endpoint health.
+            interval_in_seconds:
+                description:
+                    - The monitor interval for endpoints in this profile.
+            timeout_in_seconds:
+                description:
+                    - The monitor timeout for endpoints in this profile.
+            tolerated_number_of_failures:
+                description:
+                    - The number of consecutive failed health check before declaring an endpoint in this profile Degraded after the next failed health check.
+        default:
+            protocol: HTTP
+            port: 80
+            path: /
+    endpoints:
```

Comment:
Endpoints should be managed in another module. Have `endpoints` here take in a list of endpoint names or ids instead of the whole endpoint definition.

Reply:
fixed